src/Tools/dotnet-monitor/CollectionRules/Configuration/CollectionRuleConfigureNamedOptions.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Options;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Options;
+using System;
+using System.Diagnostics;
+using System.Globalization;
+
+namespace Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Configuration
+{
+    internal sealed class CollectionRuleConfigureNamedOptions :
+        IConfigureNamedOptions<CollectionRuleOptions>
+    {
+        private readonly ICollectionRuleActionOptionsProvider _actionOptionsProvider;
+        private readonly IConfigurationSection _collectionRulesSection;
+        private readonly ICollectionRuleTriggerOptionsProvider _triggerOptionsProvider;
+
+        public CollectionRuleConfigureNamedOptions(
+            IConfiguration configuration,
+            ICollectionRuleActionOptionsProvider actionOptionsProvider,
+            ICollectionRuleTriggerOptionsProvider triggerOptionsProvider)
+        {
+            _actionOptionsProvider = actionOptionsProvider;
+            _collectionRulesSection = configuration.GetSection(nameof(ConfigurationKeys.CollectionRules));
+            _triggerOptionsProvider = triggerOptionsProvider;
+        }
+
+        public void Configure(string name, CollectionRuleOptions options)
+        {
+            IConfigurationSection ruleSection = _collectionRulesSection.GetSection(name);
+            Debug.Assert(ruleSection.Exists());
```

Comment:
Is this assert necessary?

Reply:
It's helpful when debugging that the name you provided couldn't be found in the configuration.